packages/deployments/contracts/contracts/messaging/RootManager.sol

```diff
@@ -163,7 +170,7 @@ contract RootManager is ProposedOwnable, IRootManager, WatcherClient, DomainInde
     (bytes32 _aggregateRoot, uint256 _count) = MERKLE.insert(_verifiedInboundRoots);
 
     for (uint32 i; i < _numDomains; ) {
-      IHubConnector(_connectors[i]).sendMessage(abi.encodePacked(_aggregateRoot));
+      IHubConnector(_connectors[i]).sendMessage{value: _fees[i]}(abi.encodePacked(_aggregateRoot), _encodedData[i]);
```

Comment:
the sum value of all fees must add up to what this contract received in ETH when `propagate` was called, otherwise 1 of 2 things could happen:
- ETH could get left behind in the contract (w/ no dusting methods here)
- someone could call `propagate` and send fees to the wrong connectors as a griefing vector

Reply:
> someone could call propagate and send fees to the wrong connectors as a griefing vector

Also, this problem kind of arises here in general. With every connector, a different fee requirement will exist. Without a way to enforce that fee requirement on-chain, and with this method being external w/o accessibility restrictions, someone could grief the messaging layer by calling propagate with 0 fees provided when fees are in fact needed.

We need to either restrict this method to relayers for now OR pre-configure fees ahead of time for each connector (e.g. each connector expects a certain fee amount in the `sendMessage` call, and the owner configures that amount separately)

Reply:
fixed by decrementing the sum, should error on underflow